docs/modules/policies/pages/compile.adoc

```diff
@@ -14,45 +14,73 @@ docker run -i -t -v /path/to/policy/dir:/policies {app-docker-img} compile /poli
 
-Tests are defined using the familiar YAML format as well. Make sure that your tests are in separate directory from the policies to avoid confusion. We recommend storing them in a top-level directory named `tests`.
+Tests are defined using the familiar YAML format as well. Make sure that your tests are in separate directory from the policies to avoid confusion. We recommend storing them in a top-level directory named `tests`. A test file must have `_test` suffix in the name and one of the following file extensions: 'yaml', 'yml', or 'json'. For example, `album_test.yml`.
 
 .Test suite definition
 [source,yaml]
 ----
 ---
 name: AlbumObjectTestSuite <1>
 description: Tests for verifying the album:object resource policy <2>
-tests: <3>
-  - name: Alicia tries to view her own private album <4>
-    input: { <5>
+resources: <3>
+  alicia_album:
+    kind: "album:object",
+    attr:
+      owner: aliciaID,
+      id: XX125,
+      public: false,
+      flagged: false
+principals: <4>
+  bradley:
+    id: bradleyID
+    roles:
+      - user
+  alicia:
+    id: aliciaID
+    roles:
+      - user
+tests: <5>
+  - name: Alicia tries to view her own private album <6>
+    input: { <7>
       "requestId": "test",
-      "actions": ["view"],
-      "principal": {
-        "id": "alicia",
-        "roles": ["user"]
-      },
-      "resource": {
-        "kind": "album:object",
-        "attr": {
-          "owner": "alicia",
-          "id": "XX125",
-          "public": false,
-          "flagged": false
-        }
-      }
+      "actions": ["view", "delete"],
+      "resource": "alicia_album"
     }
-    expected: <6>
-      view: EFFECT_ALLOW
+    expected: <8>
+      - principal: alicia
+        actions:
+          view: EFFECT_ALLOW
+          delete: EFFECT_ALLOW
+      - principal: bradley
+        actions:
+          view: EFFECT_DENY
+          delete: EFFECT_DENY
 ----
 <1> Name of the test suite
 <2> Description of the test suite
-<3> List of tests in this suite
-<4> Name of the test
-<5> Input to the policy engine
-<6> Expected outcome for each of the actions defined in the input
+<3> Map of resources. A key is a string that can be used to refer to the associated resource
+<4> Map of principals. A key is a string that can be used to refer to the associated principal
+<5> List of tests in this suite
+<6> Name of the test
+<7> Input to the policy engine
+<8> List of outcome expected for a specified principal and a given action.
 
+It is possible to share principals and resources between
+test suites stored in the same directory. Create a `testdata` directory in the directory with your test suite files, then put shared resources and principals to `testdata\resources.yml` and `testdata\principals.yml` respectively (`yaml` and `json` extensions are also supported).
 
-To run the tests, provide the path to the tests directory using the `--tests` flag.
+.An example of `testdata\principals.yml`
```

Comment:
Let's add an example for `resources.yaml` as well. We can also illustrate how to use YAML anchors and link to an external site that has good documentation on the subject.

Reply:
Done